warp10/src/main/java/io/warp10/standalone/StandaloneScriptRunner.java

```diff
@@ -64,6 +65,12 @@ public class StandaloneScriptRunner extends ScriptRunner {
 
   private static final Pattern VAR = Pattern.compile("\\$\\{([^}]+)\\}");
 
+  /**
+   * runContexts < script path , hashmap > can store context objects for next runner iteration.
+   * Currently stores only runner.execution.counter, but can be extended later.
+   */
+  final Map<String, HashMap> runContexts = new ConcurrentHashMap<String, HashMap>();
```

Comment:
the runContexts map is never purged, which will lead to leaking in the case of long running Warp 10 instances with an evolving list of runners.

The map needs to be cleaned periodically, maybe after scheduling?

Reply:
Done. There was also a leak for nextrun hashmap, I fixed the problem.